content/en/docs/tasks/configure-pod-container/configure-gmsa.md

```diff
@@ -0,0 +1,302 @@
+
+In Kubernetes, GMSA credential specs are configured at a Kubernetes cluster-wide scope as custom resources. Windows pods, as well as individual containers within a pod, can be configured to use a GMSA for domain based functions (e.g. Kerberos authentication) when interacting with other Windows services. As of v1.14, the only container runtime interface that supports GMSA for Windows workloads is Dockershim. Implementation of GMSA through CRI and other runtimes is planned for the future.
+
+{{< note >}}
+Currently this feature is in alpha state. While the overall goals and functionality will not change, the way in which the GMSA credspec references are specified in pod specs may change from annotations to a API fields. Please take this into consideration when testing or adopting this feature.
+{{< /note >}}
+
+{{% /capture %}}
+
+{{% capture body %}}
+
+## Setup and configuration for GMSA
+Configuring GMSA credential specs in the cluster and configuring individual pods and containers to be able to use them requires several steps described in details below:
+1. Enable the `WindowsGMSA` feature gate on kubelet on Windows nodes.
+2. Install the GMSACredentialSpec CRD
+3. Create GMSA credential spec resources
+4. Install webhooks to expand and validate references to GMSA credential spec resources from pod specs
+5. Create cluster roles to allow service accounts to use specific GMSA credential spec resources
+6. Bind roles to specific service accounts to allow them to use GMSA credential spec resources
+7. Configure pods to use GMSA credential specs along with a service account authorized to use the GMSA credential specs
+
+### Enable the WindowsGMSA feature gate
+In the alpha state, the `WindowsGMSA` feature gate needs to be enabled on kubelet on Windows nodes. This is required to pass down the GMSA credential specs from the cluster scoped configurations to the container runtime. See [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) for an explanation of enabling feature gates. Please make sure `--feature-gates=WindowsGMSA=true` parameter exists in the kubelet.exe command line.
+
+### Install the GMSACredentialSpec CRD
+A [CustomResourceDefinition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/) (CRD) for GMSA credential spec resources needs to be configured on the cluster to define the custom resource type `GMSACredentialSpec`. Download the GMSA CRD [YAML](https://github.com/kubernetes-sigs/windows-gmsa/blob/master/admission-webhook/deploy/gmsa-webhook.yml.tpl#L131-L148) and save it as gmsa-crd.yaml.
+Next, install the CRD with `kubectl apply -f gmsa-crd.yaml`
+
+### Create GMSA credspec resources
```

Comment:
See comment above, but I really do think it would be good to cleanly separate "one-time" operations from steps needed for each GMSA deployment afterwards (this would belong to the latter).